src/components/Inputs/ComboBox/ComboBox.tsx

```diff
@@ -82,13 +88,20 @@ export const ComboBox = <T extends ComboBoxOptionRequired>(
         aria-expanded={open}
         $underlineHighlight={underlineHighlight}
         $lightBackground={lightBackground}
-        $label={!!label}
+        $shouldShowTopMargin={shouldShowTopMargin}
       >
-        {label && (
-          <Label
-            label={label}
-            htmlFor={id ? id : `amplify-combobox-${label}`}
-          />
+        {shouldShowTopMargin && (
+          // we render both labels to ensure correct positioning
+          <>
+            <Label
+              label={label}
+              htmlFor={id ? id : `amplify-combobox-${label}`}
```

Comment:
Yes I simplified it a bit

Reply:
Not quite sure how these selectors are working now that the Label component is rendering two `<span />` elements instead of there being two separate Labels on the DOM...

`  // label
  > label {
    left: 0;
    position: absolute;
    top: -1rem;
  }

  // meta
  > label {
    right: 0;
    position: absolute;
    top: -1rem;
  }`